builder/processor/src/main/java/io/helidon/builder/processor/BlueprintProcessor.java

```diff
@@ -0,0 +1,574 @@
+ * limitations under the License.
+ */
+
+package io.helidon.builder.processor;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import javax.tools.Diagnostic;
+import javax.tools.JavaFileObject;
+
+import io.helidon.builder.processor.ValidationTask.ValidateConfiguredType;
+import io.helidon.common.Errors;
+import io.helidon.common.processor.CopyrightHandler;
+import io.helidon.common.processor.GeneratedAnnotationHandler;
+import io.helidon.common.processor.TypeInfoFactory;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypeName;
+
+import static io.helidon.builder.processor.Types.FACTORY_METHOD;
+import static io.helidon.builder.processor.Types.GENERATED;
+import static io.helidon.builder.processor.Types.OVERRIDE;
+import static io.helidon.builder.processor.Types.PROTOTYPE_BLUEPRINT;
+import static io.helidon.builder.processor.Types.RUNTIME_PROTOTYPE;
+import static io.helidon.builder.processor.Types.RUNTIME_PROTOTYPE_TYPE;
+
+/**
+ * Annotation processor for prototype blueprints.
+ * Generates prototype implementation from the blueprint.
+ */
+public class BlueprintProcessor extends AbstractProcessor {
+    private static final String SOURCE_SPACING = "    ";
+    private static final TypeName GENERATOR = TypeName.create(BlueprintProcessor.class);
+    private final Set<ValidationTask> validationTasks = new LinkedHashSet<>();
+    private final Set<Element> runtimeTypes = new HashSet<>();
+    private final Set<Element> blueprintTypes = new HashSet<>();
+
+    private TypeElement blueprintAnnotationType;
+    private TypeElement runtimePrototypeAnnotationType;
+    private Messager messager;
+    private Filer filer;
+    private ProcessingEnvironment env;
+    private Elements elementUtils;
+
+    @Override
+    public Set<String> getSupportedAnnotationTypes() {
+        return Set.of(PROTOTYPE_BLUEPRINT,
+                      RUNTIME_PROTOTYPE,
+                      GENERATED);
+    }
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latestSupported();
+    }
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+
+        this.elementUtils = processingEnv.getElementUtils();
```

Comment:
I think it would be better/safer/cleaner to just get these directly from the super getters instead of holding them locally. `processingEnv` is fine, but the others not so much.

Reply:
There is no getter for elements utils (or other used instances) on the super type. Not sure what you expect me to do

Reply:
if you hold `processingEnv` as the only attribute, you can have a getter method for `getElementUtils` that uses the `processingEnv` field.  Same for the other couple attributes like this.

Reply:
There is one processing environment. Do you expect the method `getElementUtils` to return different things during tha annotation processing rounds?
Why do you mind at all? This is an implementation approach that is used in some annotation processors, and it has no ill side effects AFAIK

Reply:
Having additional attributes is unneeded, and will go stale easily after the round - and would create potential bug vectors

Reply:
They do not go stale. I am using them in the last round to analyze types that were created in previous rounds. This is a non-issue.